scene/gui/line_edit.cpp

```diff
@@ -870,6 +870,15 @@ void LineEdit::delete_char() {
 
 void LineEdit::delete_text(int p_from_column, int p_to_column) {
 
+	p_from_column = MAX(0, p_from_column);
+	p_to_column = MIN(p_to_column, text.size());
+
+	if (p_from_column >= p_to_column) {
+
+		// Nothing to do
+		return;
+	}
```

Comment:
I saw how it has been handled on master and 3.2, however I don't think it is an error it is like the job has already been done (somehow). However adding a warning might be useful for that case, but I don't want users to open new issue on github because of a warning message that hurts nobody. It'll be better to have another warning message level for developpers rather than throwing a warning message at the face of all existing users.

Reply:
Well "all existing users" for the `2.1` branch is a handful of persons :)

So it's better to be consistent with what is currently implemented in the stable release and in the development branch. If you think this shouldn't be an error this can be reviewed for `master` but it's out of scope for a `2.1` PR.

I do think that this should be an error, calling `delete_text(-42, 33)` on a "Hello world" is a bug and should be reported as such (`ERR_FAIL`). What you implemented will make it work like `delete_text(0, 11)` which is not at all what the user would expect from calling `delete_text(-42, 33)`.

Reply:
Indeed.